docs/hints-for-use-with-docker.md

```diff
@@ -0,0 +1,52 @@
+# Hints for using ORT with Docker
+
+- [Hints for using ORT with Docker](#hints-for-using-ort-with-docker)
+  - [General Hints](#general-hints)
+    - [Mount current working directory or directory relative to working directory](#mount-current-working-directory-or-directory-relative-to-working-directory)
+  - [Common Issues](#common-issues)
+    - [Docker build fails with a "SSL Handshake" error](#docker-build-fails-with-a-ssl-handshake-error)
+    - [Authenticating with a private git repository fails](#authenticating-with-a-private-git-repository-fails)
+
+## General Hints
+
+### Mount current working directory or directory relative to working directory
+
+Use `$(pwd)` on unix-based systems, `${PWD}` for PowerShell on Windows:
+
+```shell
+docker run \
+  -v $(pwd):/project  \ # mount current working directory into /project to use as input
+  -v $(pwd).ort:/ort \ # mount folder "<workingdirectory>.ort" to /ort to use as output and config file storage
+  ort --info analyze \
+  -c /ort/config.hocon \ # use file from "<workingdirectory>.ort" as config
+  analyze (...) # further options
+```
+
+## Common Issues
+
+### Docker build fails with a "SSL Handshake" error
+
+1. Insert `ENV JAVA_OPTS="-Djdk.tls.client.protocols=TLSv1.2"` in the Dockerfile, below the `FROM` line to force a specific TLS version.
+2. Run the build again, it should succeed now.
+
+### Authenticating with a private git repository fails
+
+To authenticate with a private git repository, ORT uses the (semi)standardized .netrc file. With the following steps, `.netrc` can be used with Docker.
+
+1. Create a .netrc file:
+   ```
+   machine <hostname> login <username> password <password>
+   ```
+   For example:
+   ```
+   machine github.com login mygituser password mypersonaltoken
+   ```
+   Ensure that there are no additional spaces, newlines, tabs, or other whitespace characters after the hostname, username or password. These characters would be appended to the information and cause authentication errors for some providers.
```

Comment:
Let's not document this limitation but address it right away: https://github.com/oss-review-toolkit/ort/pull/3150

Reply:
If I recall correctly ORT is both manually parsing the .netrc (which you addressed here), and calling native git directly, which will also parse it. I also encountered this limitation with native git (latest version) on Ubuntu 20.04 when ORT was not involved at all. Correct me if you are calling git with the already parsed credentials, then this should not be an issue after your fix.

Reply:
You're correct, for tools that have native support for `.netrc` we're leaving the parsing to them, so we indeed have to align on the "lowest common denominator" regarding what syntax is accepted. But could we then reword the sentence a bit to make it sound less it it was ORT's fault? 😀 Like by rewording the second sentence to "Some third-party tools would otherwise interpret the whitespaces as part of the strings and cause ..."

Reply:
Absolutely, let's lay blame where it is due ;)